hy/compiler.py

```diff
@@ -1216,20 +1224,69 @@ def _compile_import(expr, module, names=None, importer=asty.Import):
 
         return rimports
 
+    def _compile_subscript(self, root, value, exprs):
+        dims = []
+        ret = Result()
+
+        for expr in exprs:
+            if expr_startswith(HyKeyword(":"), expr):
+                if len(expr) > 4:
+                    raise HyTypeError(expr, "slice expected at most "
+                                            "3 arguments")
+
+                nodes = [None] * 3
+                for i, e in enumerate(expr[1:]):
+                    ret += self.compile(e)
+                    nodes[i] = ret.force_expr
+
+                dims.append(
+                    ast.Slice(lower=nodes[0], upper=nodes[1], step=nodes[2]))
+            elif not PY3 and isinstance(expr, HySymbol) and expr == "...":
+                dims.append(ast.Ellipsis())
+            else:
+                ret += self.compile(expr)
+                dims.append(ret.force_expr)
+
+        def box_as_slice(expr):
+            if isinstance(expr, ast.slice):
+                return expr
+            return ast.Index(value=expr)
+
+        # How the subscript is constructed will depend on if any of
+        # the components subclass ast,slice. If they do, then we have
+        # to construct an ExtSlice instead of a Tuple.
+        is_ext_slice = any(isinstance(dim, ast.slice) for dim in dims)
+
+        if len(dims) == 1:
+            index = dims[0]
+        elif is_ext_slice:
+            dims = [box_as_slice(dim) for dim in dims]
+            index = asty.ExtSlice(root, dims=dims)
+        else:
+            index = asty.Tuple(root, elts=dims, ctx=ast.Load())
+
+        return ret + asty.Subscript(
+            root,
+            value=value,
+            slice=box_as_slice(index),
+            ctx=ast.Load())
+
     @builds("get")
     @checkargs(min=2)
-    def compile_index_expression(self, expr):
-        expr.pop(0)  # index
+    def compile_index_expression(self, exprs):
```

Comment:
The reason we call this sort of parameter `expr` rather than `exprs` is because it's set to a single `HyExpression`.